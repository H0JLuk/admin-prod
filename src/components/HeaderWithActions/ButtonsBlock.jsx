import React from 'react';
import { Button } from 'antd';

const ButtonsBlock = ({ buttons, params }) => {
    return (buttons || []).map((button) => (
        <Button
            key={ button.label }
            type={ button.type }
            onClick={ button.onClick }
            disabled={ typeof button.disabled === 'function' ? button.disabled(params) : button.disabled }
        >
            { button.label }
        </Button>
    ));
};

export default ButtonsBlock;
