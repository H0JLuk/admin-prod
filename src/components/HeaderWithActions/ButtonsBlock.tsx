import React from 'react';
import { Button, ButtonProps as AntButtonProps } from 'antd';

export type ButtonProps = Omit<AntButtonProps, 'disabled' | 'label'> & {
    label: string;
    disabled?: ((params?: Record<string, string | number>) => boolean) | boolean;
};

type ButtonsBlockProps = {
    buttons: ButtonProps[];
    params?: Record<string, string | number>;
};

const ButtonsBlock: React.FC<ButtonsBlockProps> = ({ buttons, params }) =>
    (buttons || []).map(({ label, disabled, ...restProps }) => (
        <Button
            key={label}
            disabled={typeof disabled === 'function' ? disabled(params) : disabled}
            {...restProps}
        >
            {label}
        </Button>
    )) as Exclude<React.ReactNode, null | undefined | boolean | React.ReactFragment>;

export default ButtonsBlock;
