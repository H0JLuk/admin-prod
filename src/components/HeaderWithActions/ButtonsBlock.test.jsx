import React from 'react';
import { mount } from 'enzyme';

import ButtonsBlock from './ButtonsBlock';

const TEST_BUTTONS_PROPS = [
    {
        label: 'TEST_TEXT_ONE',
        type: 'TEST_PRIMARY_TYPE',
        onClick: jest.fn(),
        disabled: false,
    },
    {
        label: 'TEST_TEXT_TWO',
        type: 'TEST_DANGER_TYPE',
        onClick: jest.fn(),
        disabled: true,
    },
];

const TEST_PROPS = {
    buttons: TEST_BUTTONS_PROPS,
    params: false
};

describe('<ButtonsBlock /> tests.', () => {
    const buttonBlock = mount(<ButtonsBlock { ...TEST_PROPS } />);
    const buttons = buttonBlock.find('button');

    it('should mount buttons', () => {
        expect(buttonBlock.html()).toMatchSnapshot();
        expect(buttons).toHaveLength(TEST_BUTTONS_PROPS.length);
    });

    it('should click to button', () => {
        buttons.at(0).simulate('click');
        expect(TEST_BUTTONS_PROPS[0].onClick).toHaveBeenCalledTimes(1);
    });

    it('should not click to disabled button', () => {
        buttons.at(1).simulate('click');
        expect(TEST_BUTTONS_PROPS[1].onClick).toHaveBeenCalledTimes(0);
    });
});
