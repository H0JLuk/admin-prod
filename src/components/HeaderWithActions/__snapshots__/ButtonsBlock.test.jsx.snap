// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonsBlock /> tests. should mount buttons 1`] = `"<button type=\\"button\\" class=\\"ant-btn ant-btn-TEST_PRIMARY_TYPE\\"><span>TEST_TEXT_ONE</span></button><button disabled=\\"\\" type=\\"button\\" class=\\"ant-btn ant-btn-TEST_DANGER_TYPE\\"><span>TEST_TEXT_TWO</span></button>"`;
